pkg/controllers/nodeclaim/garbagecollection/controller.go

```diff
@@ -95,7 +95,7 @@ func (c *Controller) Reconcile(ctx context.Context) (reconcile.Result, error) {
 		if node != nil && nodeutils.GetCondition(node, corev1.NodeReady).Status == corev1.ConditionTrue {
 			return
 		}
-		if err := c.kubeClient.Delete(ctx, nodeClaims[i]); err != nil {
+		if err := c.updateStatusAndDelete(ctx, nodeClaims[i]); err != nil {
```

Comment:
nit: Could you consider doing this inline and add it above like you did the other ones -- if it fires the gocyclo linter, it's fine to disable it for the function